JSIL.Libraries/Sources/JSIL.Core.js

```diff
@@ -10732,4 +10739,207 @@ JSIL.Array.IndexOf = function (array, startIndex, count, value) {
+  this.MethodInfoResolved = false;
+  this.MethodInfo = null;
+
+  var useGenericSuffix = this.MethodGenericParameters !== null && this.MethodGenericParameters.length > 0;
+
+  this.NameWithGenericSuffix = useGenericSuffix ? (name + "$b" + this.MethodGenericParameters.length) : name;
+}
+
+JSIL.MethodInfoMethodPointerInfo = function(methodInfo) {
+  var signature = methodInfo._data.signature;
+  var genericArgs = null;
+  if (signature.openSignature !== null) {
+    genericArgs = signature.genericArgumentValues;
+    signature = signature.openSignature;
+  }
+
+  JSIL.MethodPointerInfo.call(this,
+    methodInfo._typeObject.__PublicInterface__,
+    methodInfo._descriptor.Name,
+    signature,
+    methodInfo._descriptor.Static,
+    methodInfo._descriptor.Virtual,
+    genericArgs);
+
+  this.MethodInfo = methodInfo;
+  this.MethodInfoResolved = true;
+}
+
+JSIL.MethodInfoMethodPointerInfo.prototype = JSIL.MethodPointerInfo.prototype;
+
+JSIL.MethodPointerInfo.prototype.resolveMethodInfo = function () {
+  if (!this.MethodInfoResolved) {
+    this.MethodInfoResolved = true;
+    this.MethodInfo = JSIL.GetMethodInfo(this.TypeObject, this.Name, this.Signature, this.IsStatic, this.MethodGenericParameters);
+  }
+
+  return this.MethodInfo;
+}
+
+JSIL.MethodPointerInfo.prototype.createInvocationFunction = function(thisObject) {
+  var argsCount = JSIL.IsArray(this.Signature.argumentTypes) ? this.Signature.argumentTypes.length : 0;
+
+  if (thisObject === null) {
+    if (this.TypeObject.__Type__.IsInterface) {
+      return JSIL.MethodPointerInfo.$createVirtualInterfaceInvocation(argsCount, false)(this);
+    } else if (this.IsStatic) {
+      return JSIL.MethodPointerInfo.$createStaticInvocation(argsCount, false)(this);
+    } else if (!this.IsVirtual) {
+      return JSIL.MethodPointerInfo.$createInstanceInvocation(argsCount, false)(this);
+    } else if (!this.TypeObject.__Type__.IsInterface) {
+      return JSIL.MethodPointerInfo.$createVirtualInvocation(argsCount, false)(this);
+    }
+  } else {
+    if (this.TypeObject.__Type__.IsInterface) {
+      return JSIL.MethodPointerInfo.$createVirtualInterfaceInvocation(argsCount, true)(this, thisObject);
+    } else if (this.IsStatic) {
+      return JSIL.MethodPointerInfo.$createStaticInvocation(argsCount, true)(this, thisObject);
+    } else if (!this.IsVirtual) {
+      return JSIL.MethodPointerInfo.$createInstanceInvocation(argsCount, true)(this, thisObject);
+    } else if (!this.TypeObject.__Type__.IsInterface) {
+      return JSIL.MethodPointerInfo.$createVirtualInvocation(argsCount, true)(this, thisObject);
+    }
+  }
+}
+
+JSIL.MethodPointerInfo.$createStaticInvocation = function (argsCount, saveThis)
+{
+  var key = "InvokeStatic" + (saveThis ? "Attached" : "") + argsCount;
+
+  if (!(key in JSIL.MethodPointerInfo)) {
+    var innerArgs = [];
+    var innerBody = [];
+
+    innerBody.push("//# sourceURL=jsil://closure/JSIL.MethodPointerInfo." + key + "\r\n");
```

Comment:
I need here outer function that will store closure variable. So let's see this:
```
function outer(closureArg) {
  return function inner(arg) { return closureArg + arg;};
}
```
If I call `CreateRebindableNamedFunction` it will still return me `inner` function. I need get `outer` function here, so that I'm able to reuse it.